### PR TITLE
hal-hardware-analyzer: 3.1.9 -> 3.2.5

### PR DESCRIPTION
**Pull request by SCOTT-HAMILTON**
###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by fabaff** [comment]
Result of `nixpkgs-review pr 112414` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>hal-hardware-analyzer</li>
  </ul>
</details>

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 112414` at ac4a2759 run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package marked as broken and skipped:</summary>
  <ul>
    <li>hal-hardware-analyzer</li>
  </ul>
</details>

**Review by fabaff — APPROVED** [review]
- [x] package build (see https://github.com/NixOS/nixpkgs/pull/112414#issuecomment-775318137)
- [ ] all depending packages build
- [x] tested on x86_64

```bash
...ew/pr-112414]$ ./results/hal-hardware-analyzer/bin/hal -v
3.2.5
...ew/pr-112414]$ ./results/hal-hardware-analyzer/bin/hal -g
[...]
```

GUI is coming up. Looks good to me.

**Comment by risicle** [comment]
Testing macos...

File: pkgs/applications/science/electronics/hal-hardware-analyzer/default.nix [hunk 1]
```diff
@@ -1,17 +1,18 @@
 { lib, stdenv, fetchFromGitHub, cmake, ninja, pkg-config, python3Packages
 , boost, rapidjson, qtbase, qtsvg, igraph, spdlog, wrapQtAppsHook
 , fmt, graphviz, llvmPackages ? null
```
Comment (SuperSandro2000) [inline pkgs/applications/science/electronics/hal-hardware-analyzer/default.nix:3]:
```suggestion
, fmt, graphviz, llvmPackages
```

**Review by risicle — APPROVED** [review]
WFM macos 10.14

File: pkgs/applications/science/electronics/hal-hardware-analyzer/default.nix [hunk 2]
```diff
@@ -1,17 +1,18 @@
 { lib, stdenv, fetchFromGitHub, cmake, ninja, pkg-config, python3Packages
 , boost, rapidjson, qtbase, qtsvg, igraph, spdlog, wrapQtAppsHook
-, fmt, graphviz, llvmPackages ? null
+, fmt, graphviz, llvmPackages
+, z3
```
Comment (SuperSandro2000) [inline pkgs/applications/science/electronics/hal-hardware-analyzer/default.nix:4]:
```suggestion
, fmt, graphviz, llvmPackages, z3
```

**Pull request merged**